scripts-endpoint.md

```diff
@@ -0,0 +1,169 @@
+      {
+        "name" : "metadata-import",
+        "description" : "Import metadata after batch editing"
+        "_links" : {
+          "self" : {
+            "href" : "/api/admin/scripts/metadata-import"
+          }
+        }
+      },
+      ...
+    ]
+  }
+```
+
+## Script Name Endpoint
+
+### Script Details
+**GET /api/admin/scripts/<:script-name>**
+
+This endpoint will return information on all the parameters that are required to invoke the script. This dependens on the parameters defined in the implementation of the script.
+
+The JSON response document is as follows
+```
+{
+   "name" : "import",
+   "description" : "Import items into DSpace",
+   "parameters" : [
+    {
+      "name" : "a",
+      "description" : "add items to DSpace"
+      "type" : "boolean"
+    },
+    {
+      "name" : "b",
+      "description" : "add items to DSpace via Biblio-Transformation-Engine (BTE)"
+      "type" : "boolean"
+    },
+    {
+      "name" : "c",
+      "description" : "destination collection(s) database ID"
+      "type" : "id"
+    },
+    {
+      "name" : "d",
+      "description" : "delete items listed in mapfile"
+      "type" : "file"
+    },
+    {
+      "name" : "e",
+      "description" : "email of eperson doing importing"
+      "type" : "string"
+    },
+    {
+      "name" : "i",
+      "description" : "input type in case of BTE import"
+      "type" : "string"
+    },
+    {
+      "name" : "n",
+      "description" : "if sending submissions through the workflow, send notification emails"
+      "type" : "boolean"
+    },
+    {
+      "name" : "p",
+      "description" : "apply template"
+      "type" : "string"
+    },
+    {
+      "name" : "q",
+      "description" : "don't display metadata"
+      "type" : "boolean"
+    },
+    {
+      "name" : "m",
+      "description" : "mapfile items in mapfile"
+      "type" : "file"
+    },
+    {
+      "name" : "r",
+      "description" : "replace items in mapfile"
+      "type" : "boolean"
+    },
+    {
+      "name" : "R",
+      "description" : "resume a failed import (add only)"
+      "type" : "boolean"
+    },
+    {
+      "name" : "t",
+      "description" : "test run - do not actually import items"
+      "type" : "boolean"
+    },
+    {
+      "name" : "w",
+      "description" : "send submission through collection's workflow"
+      "type" : "boolean"
+    },
+    {
+      "name" : "z",
+      "description" : "zip file containig the import"
+      "type" : "file"
+    }
+   ]
+}
+```
+
+Following parameter types are available:
+* `string`: A regular string value
+* `date`: A string that has a valid date pattern
+* `boolean`: true or false
+* `file`: For parameters with this type, the user has to provide a file
+* `output`: Parameters with this type define the name of the output file. This name can be used later to download the the output file (e.g. when running `export` or `metadata-export`).
+
+### Script Invocation
+**POST /api/admin/scripts/<:script-name>**
```

Comment:
The POST should be issued against the processes endpoint. You can create a process just giving the link to the script to execute and the information about the execution environments (variables, etc.)

Reply:
I agree that the POST should happen on the processes endpoint. I'll think about how we can support this in the contract.

Reply:
Sorry I don't understand the question. Is it about how to format the documentation or how to arrange the json body?
we have started to add documentation about additional verbs at the end of the page see https://github.com/4Science/Rest7Contract/blob/DS-3759/workspaceitems.md#multipart-post-method we can always rearrange that later, no a strong opinion about the best formatting (and we need to keep in mind that our manual REST contract should be replaced at some point by something semi-automatically produced see )

if you look for how to arrange the json body
`{ script: "https://..../api/config/scripts/import', parameters: { ... }}`